pal/zephyros/pal_zephyr.h

```diff
@@ -0,0 +1,99 @@
+/**
+* \copyright
+* MIT License
+*
+* Copyright (c) 2019 Infineon Technologies AG
+*
+* Permission is hereby granted, free of charge, to any person obtaining a copy
+* of this software and associated documentation files (the "Software"), to deal
+* in the Software without restriction, including without limitation the rights
+* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+* copies of the Software, and to permit persons to whom the Software is
+* furnished to do so, subject to the following conditions:
+*
+* The above copyright notice and this permission notice shall be included in all
+* copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+* SOFTWARE
+*
+* \endcopyright
+*
+* \author Infineon Technologies AG
+*
+* \file pal_gpio.c
+*
+* \brief   This file provides the prototype declarations w.r.t Zephyr OS.
+*
+* \ingroup  grPAL
+*
+* @{
+*/
+
+#ifndef _PAL_ZEPHYR_H
+#define _PAL_ZEPHYR_H
+
+/* OPTIGA includes */
+#include "optiga/pal/pal.h"
+
+/* TODO: Make defines not static but depending on KCONFIG */
+#define I2C_OPTIGA_ADDRESS  CONFIG_TRUSTM_I2C_ADDR
+
+
+/* I2C PORT to be used to connection */
+#define I2C_PORT_NAME     CONFIG_TRUSTM_I2C_PORT_NAME // DT_ALIAS_I2C_0_LABEL
+/* I2C_0 pins on nRF52840_pca10056 */
+#define I2C_SDA     26 /* 26 */
+#define I2C_SCL     27 /* 27 */
```

Comment:
Can't this be defined from the configuration menu for zephyr? I mean if not defined use some default value, otherwise what has been selected

Reply:
These can defined with options from the configuration menu too. From your second statement I derive that you suggest to put these defines in #ifdef/#ifndef blocks just to make sure that that these are always well defined.